src/react/hooks/useSuspenseQuery.ts

```diff
@@ -196,29 +196,25 @@ export function useSuspenseQuery<
     client.watchQuery(watchQueryOptions)
   );
 
-  const [promiseCache, setPromiseCache] = React.useState(
-    () => new Map([[queryRef.key, queryRef.promise]])
-  );
-
-  let promise = promiseCache.get(queryRef.key);
+  let [current, setPromise] = React.useState<
+    [QueryKey, Promise<ApolloQueryResult<any>>]
+  >([queryRef.key, queryRef.promise]);
```

Comment:
```suggestion
  >(() => [queryRef.key, queryRef.promise]);
```

While very minor, would it make sense to avoid allocating new memory for unused arrays on each render?